WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift

```diff
@@ -0,0 +1,309 @@
+    }
+
+    func configureCountry(cell: TitleAndTextFieldTableViewCell, row: Row) {
+        let viewModel = TitleAndTextFieldTableViewCell.ViewModel(title: Localization.countryField,
+                                                                 text: "",
+                                                                 placeholder: Localization.countryFieldPlaceholder,
+                                                                 state: .normal,
+                                                                 keyboardType: .default,
+                                                                 textFieldAlignment: .leading) { (newText) in
+
+        }
+        cell.configure(viewModel: viewModel)
+    }
+}
+
+// MARK: KeyboardScrollable
+extension ShippingLabelAddressValidationViewController: KeyboardScrollable {
+    var scrollable: UIScrollView {
+        tableView
+    }
+}
+
+extension ShippingLabelAddressValidationViewController {
+
+    struct Section: Equatable {
+        let rows: [Row]
+    }
+
+    enum Row: CaseIterable {
+        case topBanner
+
+        case name
+        case company
+        case phones
+        case address
+        case address2
+        case city
+        case postcode
+        case state
+        case country
+
+        case fieldError
+
+        fileprivate var type: UITableViewCell.Type {
+            switch self {
+            case .topBanner, .fieldError:
+                return BasicTableViewCell.self
+            case .name, .company, .phones, .address, .address2, .city, .postcode, .state, .country:
+                return TitleAndTextFieldTableViewCell.self
+            }
+        }
+
+        fileprivate var reuseIdentifier: String {
+            return type.reuseIdentifier
+        }
+    }
+}
+
+private extension ShippingLabelAddressValidationViewController {
+    enum Localization {
+        static let titleViewShipFrom = NSLocalizedString("Ship from", comment: "Shipping Label Address Validation navigation title")
+        static let titleViewShipTo = NSLocalizedString("Ship to", comment: "Shipping Label Address Validation navigation title")
+
+        static let nameField = NSLocalizedString("Name", comment: "Text field name in Shipping Label Address Validation")
+        static let nameFieldPlaceholder = NSLocalizedString("Required", comment: "Text field placeholder in Shipping Label Address Validation")
+        static let companyField = NSLocalizedString("Company", comment: "Text field company in Shipping Label Address Validation")
+        static let companyFieldPlaceholder = NSLocalizedString("Optional", comment: "Text field placeholder in Shipping Label Address Validation")
+        static let phonesField = NSLocalizedString("Phones", comment: "Text field phones in Shipping Label Address Validation")
+        static let phonesFieldPlaceholder = NSLocalizedString("Optional", comment: "Text field placeholder in Shipping Label Address Validation")
```

Comment:
nit: similar to above, maybe `phoneField*`?
```suggestion
        static let phoneField = NSLocalizedString("Phones", comment: "Text field phones in Shipping Label Address Validation")
        static let phoneFieldPlaceholder = NSLocalizedString("Optional", comment: "Text field placeholder in Shipping Label Address Validation")
```